crates/aggchain-proof-core/src/full_execution_proof.rs

```diff
@@ -240,16 +271,17 @@ impl FepInputs {
     }
 
     /// Compute l2 pre root.
-    pub fn compute_l2_pre_root(&self) -> Digest {
+    pub fn compute_l2_pre_root(&self) -> L2PreRoot {
         compute_output_root(
             self.prev_state_root.0,
             self.prev_withdrawal_storage_root.0,
             self.prev_block_hash.0,
         )
+        .into()
     }
 
     /// Compute claim root.
-    pub fn compute_claim_root(&self) -> Digest {
+    pub fn compute_claim_root(&self) -> OutputRoot {
```

Comment:
nit: I think `L2PreRoot` is also an `OutputRoot`. If we want to bring distinct type to help us differentiate between `l2_pre_root` and `claim_root` (aka. `l2_post_root`). Then I would use `L2PreRoot` and `ClaimRoot` instead

```suggestion
    pub fn compute_claim_root(&self) -> ClaimRoot {
```

(or `L2PostRoot` but then need rename the functions and other `claim_root` occurrences)

Reply:
I wasn't sure about the naming for those 2, I'm fine switching to `ClaimRoot`